chain/types/signedmessage.go

```diff
@@ -81,6 +82,45 @@ func (sm *SignedMessage) Size() int {
 	return len(serdata)
 }
 
+func (sm *SignedMessage) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
+		return xerrors.Errorf("expected at least 1 byte")
+	}
+
+	switch b[0] {
+	case '"':
+		var b []byte
+		if err := json.Unmarshal(b, &b); err != nil {
+			return err
+		}
+
+		dm, err := DecodeSignedMessage(b)
+		if err != nil {
+			return err
+		}
+
+		*sm = *dm
+
+		return nil
+	case '{':
+		msg := struct {
+			Message   Message
+			Signature crypto.Signature
+		}{}
+
+		if err := json.Unmarshal(b, &msg); err != nil {
+			return err
+		}
```

Comment:
You can use local typedef here to disable the Unmarshal.
It would look like this:
```go
type CopySignedMesage SignedMessage
cpy := (*CopySignedMesage)(sm)
json.Unmarshal(b, cpy)
```